test/jdk/java/net/vthread/InterruptHttp.java

```diff
@@ -0,0 +1,121 @@
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @summary Test that HTTP connections can be interrupted
+ * @compile --enable-preview -source ${jdk.version} InterruptHttp.java
+ * @run main/othervm --enable-preview InterruptHttp
+ */
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class InterruptHttp {
+
+    public static void main(String[] args) throws Exception {
+        try (Server server = new Server()) {
+            AtomicReference<Exception> exception = new AtomicReference<>();
+
+            // start thread to connect to server
+            Thread thread = Thread.ofVirtual().start(() -> {
+                try {
+                    URL url = new URL(server.address() + "/hello");
+                    try (InputStream in = url.openStream()) { }
+                } catch (Exception e) {
+                    exception.set(e);
+                }
+            });
+
+            // give time for thread to block waiting for HTTP server
+            while (server.connectionCount() == 0) {
+                Thread.sleep(100);
+            }
+            thread.interrupt();
+            thread.join();
+
+            // HTTP request should fail with a SocketException
+            Exception e = exception.get();
+            if (!(e instanceof SocketException))
+                throw new RuntimeException("Expected SocketException, got: " + e);
+
+            // server should have accepted one connection
+            if (server.connectionCount() != 1)
+                throw new RuntimeException("Expected 1 HTTP connection");
+        }
+    }
+
+    /**
+     * Simple server that accepts connections but does not reply.
+     */
+    static class Server implements Closeable {
+        private final ServerSocket listener;
+        private final List<Socket> connections = new CopyOnWriteArrayList<>();
+        private final String address;
+
+        Server() throws IOException {
+            InetAddress lb = InetAddress.getLoopbackAddress();
+            listener = new ServerSocket(0, -1, lb);
+            address = "http://" + lb.getHostAddress() + ":" + listener.getLocalPort();
```

Comment:
Same remark about using URIBuilder here. It would take care of properly formatting the address in case of IPv6 literals. This can be taken care of after this PR has been integrated.